flocker/node/functional/test_docker.py

```diff
@@ -205,6 +205,38 @@ def test_dead_is_listed(self):
                                  expected_states=(u'inactive',))
         return d
 
+    def test_list_with_missing_image(self):
+        """
+        ``DockerClient.list()`` can list containers whose image is missing.
+
+        The resulting output may be inaccurate, but that's OK: this only
+        happens for non-running containers, who at worst we're going to
+        restart anyway.
+        """
+        path = FilePath(self.mktemp())
+        path.makedirs()
+        path.child(b"Dockerfile.in").setContent("FROM busybox\n")
```

Comment:
This `Dockerfile` appears to be insufficient to actually create a new image.  What it actually does is create a new tag of the busybox image!  This exercises a different case than you intended, I believe (as long as there are multiple tags of an image it seems Docker is always happy to let you remove one of them, even if a running container is using it).  I think you need at least one more statement to create at least one layer so you get a distinct image.